cl_dll/vgui_UnicodeTextImage.cpp

```diff
@@ -0,0 +1,558 @@
+	virtual int ConvertANSIToUnicode(const char *ansi, wchar_t *unicode, int unicodeBufferSizeInBytes) = 0;
+};
+
+#define VGUI_LOCALIZE_INTERFACE_VERSION "VGUI_Localize003"
+
+bool g_bSurfaceLoaded = false;
+
+// hw.dll or sw.dll on Windows, hw.so on Linux, hw.dylib on macOS.
+CSysModule *g_hEngineModule = nullptr;
+CSysModule *g_hVGuiModule = nullptr;
+
+ISurface *g_pVGuiSurface = nullptr;
+ILocalize *g_pVGuiLocalize = nullptr;
+
+}
+
+void UnicodeTextImage::initInterfaces()
+{
+	static bool bTriedToLoad = false;
+	if (bTriedToLoad)
+		return;
+
+	// Surface needs to be loaded only once, even if it fails.
+	bTriedToLoad = true;
+
+#ifdef _WIN32
+
+	HMODULE hw, sw, vgui2;
+
+	// Try hardware engine (hw.dll)
+	GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, "hw.dll", &hw);
+	if (hw)
+	{
+		g_hEngineModule = reinterpret_cast<CSysModule *>(hw);
+	}
+	else
+	{
+		// Try software engine (sw.dll)
+		GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, "hw.dll", &sw);
+		if (sw)
+		{
+			g_hEngineModule = reinterpret_cast<CSysModule *>(sw);
+		}
+	}
+
+	// Load vgui2.dll
+	GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, "vgui2.dll", &vgui2);
+	g_hVGuiModule = reinterpret_cast<CSysModule *>(vgui2);
+
+#else
+	
+#if defined(OSX)
+#define DLL_EXT ".dylib"
+#elif defined(LINUX)
+#define DLL_EXT ".so"
+#endif
+
+	// hw.so is guranteed to be loaded as it's the only engine library on Linux/macOS.
+	g_hEngineModule = reinterpret_cast<CSysModule *>(dlopen("./hw" DLL_EXT, RTLD_NOW | RTLD_NOLOAD));
+	g_hVGuiModule = reinterpret_cast<CSysModule *>(dlopen("./vgui2" DLL_EXT, RTLD_NOW | RTLD_NOLOAD));
+
+#endif
+
+	if (!g_hEngineModule)
+	{
+		ConsolePrint("UnicodeTextImage: Error: Engine module handle is null.\n");
+		return;
+	}
+
+	if (!g_hVGuiModule)
+	{
+		ConsolePrint("UnicodeTextImage: Error: VGUI2 library handle is null.\n");
+		return;
+	}
+
+	auto fnLoadInterface = [](const char *moduleName, CSysModule *pModule, const char *ifaceName, auto &pIface)
+	{
+		using IfaceType = typename std::remove_reference<decltype(pIface)>::type;
+
+		CreateInterfaceFn fnFactory = Sys_GetFactory(pModule);
+		if (!fnFactory)
+		{
+			gEngfuncs.Con_Printf("UnicodeTextImage: Error: %s doesn't export factory.\n", moduleName);
+			return false;
+		}
+
+		pIface = static_cast<IfaceType>(fnFactory(ifaceName, nullptr));
+		if (!g_pVGuiSurface)
+		{
+			gEngfuncs.Con_Printf("UnicodeTextImage: Error: %s doesn't export %s.\n", moduleName, ifaceName);
+			return false;
+		}
+
+		return true;
+	};
+
+	if (!fnLoadInterface("Engine", g_hEngineModule, VGUI_SURFACE_INTERFACE_VERSION, g_pVGuiSurface))
+		return;
+
+	if (!fnLoadInterface("VGUI2", g_hVGuiModule, VGUI_LOCALIZE_INTERFACE_VERSION, g_pVGuiLocalize))
+		return;
+
+	g_bSurfaceLoaded = true;
+}
+
+bool UnicodeTextImage::shouldFallback()
+{
+	return !g_bSurfaceLoaded;
+}
+
+UnicodeTextImage::HFont UnicodeTextImage::createFont(const char *fontName, int tall, int weight, bool antialias)
```

Comment:
The `antialias` argument can be removed now that the cvars are removed.

Reply:
I removed it.